spec/modules.md

```diff
@@ -0,0 +1,428 @@
+
+* If `url` ends in `.scss`, `.sass`, or `.css`:
+
+  * If this algorithm is being run for an `@import`:
+
+    * Let `suffix` be the trailing `.scss`, `.sass`, `.css` in `url`, and
+      `prefix` the portion of `url` before `suffix`.
+
+    * If the result of [resolving `prefix` + `".import"` + `suffix` for
+      partials][resolving for partials] is not null, return it.
+
+  * Otherwise, return the result of [resolving `url` for partials][resolving for
+    partials].
+
+  > `@import`s whose URLs explicitly end in `.css` will have been treated as
+  > plain CSS `@import`s before this algorithm even runs, so `url` will only end
+  > in `.css` for `@use` rules.
+
+* If this algorithm is being run for an `@import`:
+
+  * Let `sass` be the result of [resolving `url` + `".import.sass"` for
+    partials][resolving for partials].
+
+  * Let `scss` be the result of [resolving `url` + `".import.scss"` for
+    partials][resolving for partials].
+
+  * If neither `sass` nor `scss` are null, throw an error.
+
+  * Otherwise, if exactly one of `sass` and `scss` is null, return the other
+    one.
+
+  * Otherwise, if the result of [resolving `url` + `".import.css"` for
+    partials][resolving for partials] is not null, return it.
+
+* Otherwise, let `sass` be the result of [resolving `url` + `".sass"` for
+  partials][resolving for partials].
+
+* Let `scss` be the result of [resolving `url` + `".scss"` for
+  partials][resolving for partials].
+
+* If neither `sass` nor `scss` are null, throw an error.
+
+* Otherwise, if exactly one of `sass` and `scss` is null, return the other
+  one.
+
+* Otherwise, return the result of [resolving `url` + `".css"` for
+  partials][resolving for partials]. .
+
+[resolving for partials]: #resolving-a-file-url-for-partials
+
+### Resolving a `file:` URL for Partials
+
+This algorithm takes a URL, `url`, whose scheme must be `file` and returns
+either another URL that's guaranteed to point to a file on disk or null.
+
+* If `url`'s [basename](#basename) begins with `"_"`:
+
+  * If a file exists on disk at `url`, return `url`.
+
+    Otherwise return null.
+
+* Let `partial` be [`dirname(url)`](#dirname) + `"_"` +
+  [`basename(url)`](#basename).
+
+* If a file exists on disk at both `url` and `partial`, throw an error.
+
+* If a file exists on disk at `url`, return `url`.
+
+* If a file exists on disk at `partial`, return `partial`.
+
+* Return null.
+
+### Resolving a Member
+
+This algorithm takes a [member](#member) name `name` and a member type `type`,
+and returns a member of type `type` or null.
+
+* If `name` is a plain `Identifier` or a `Variable` that's not a
+  `NamespacedVariable`:
+
+  * Let `scope` be the [scope][] of the innermost block containing the current
+    statement such that `scope` has a member of type `type` named `name`, or
+    null if no such scope exists.
+
+    [scope]: variables.md#scope
+
+  * If `scope` is not null, return `scope`'s value of type `type` named `name`.
```

Comment:
What happens if `scope` is null here?

Reply:
We fall through to the next line. The idea is that it works like a programming language:

```
if (name is a plain identifier or variable) {
  scope = the scope of the innermost block etc etc
  if (scope != null) return scope;
}
```